app/models/task_file.rb

```diff
@@ -6,6 +6,8 @@ class TaskFile < ApplicationRecord
   has_one_attached :attachment
   validates :name, presence: true
   validates :attachment, presence: true, if: -> { use_attached_file == 'true' }, on: :force_validations
+  validates :xml_id, presence: true
+  validates :xml_id, uniqueness: {scope: %i[fileable_id fileable_type]}
```

Comment:
At least in my tests, this validation doesn't work as expected. I was able to add two new files with the same XML ID and save those. Only when editing, this validation kicks in and prevents saving the task. Do we skip the validation for new task_files for some reason?

<img width="1125" alt="Bildschirm­foto 2022-12-17 um 10 29 32" src="https://user-images.githubusercontent.com/7300329/208235475-079fb0ed-cc66-4f65-8e18-c476b676e9b7.png">
<img width="601" alt="Bildschirm­foto 2022-12-17 um 10 30 25" src="https://user-images.githubusercontent.com/7300329/208235476-89e942b5-4f77-4540-a3f0-b4764a541877.png">

Reply:
Additionally, the scope is probably not sufficient yet. The following entry is valid according to the validation:

<img width="607" alt="Bildschirm­foto 2022-12-17 um 10 45 20" src="https://user-images.githubusercontent.com/7300329/208235982-58146a9a-ff01-4523-ad66-fe2d758b99fe.png">

but generates an invalid XML:

<img width="788" alt="Bildschirm­foto 2022-12-17 um 10 44 45" src="https://user-images.githubusercontent.com/7300329/208235998-125b7308-3360-4801-bd0f-a015bc1722bd.png">

```
Proforma::PostGenerateValidationError - [#<Nokogiri::XML::SyntaxError: 36:0: ERROR: Element '{urn:proforma:v2.0.1}file': Duplicate key-sequence ['0'] in key identity-constraint '{urn:proforma:v2.0.1}task-fileids'.>]
```

Reply:
I added a custom validation that should cover this